File: src/main/java/com/qmetry/QTMGradlePlugin.java
package com.qmetry;

import org.gradle.api.Plugin;
import org.gradle.api.Project;

public class QTMGradlePlugin implements Plugin<Project> 
{
    @Override
    public void apply(Project project) 
	{
        project.getExtensions().create("qmetryConfig", QTMGradleExtension.class);
        project.getTasks().create("publishResults", PublishResults.class);
    }
}

File: src/main/kotlin/com/qmetry/QTMGradlePlugin.kt
package com.qmetry

import org.gradle.api.Plugin
import org.gradle.api.Project

class QTMGradlePlugin : Plugin<Project> {
    override fun apply(project: Project) {
        project.extensions.create("qtmConfig", QTMGradleExtension::class.java)
        project.tasks.create("qtmUploadResults", QTMResultsPublisher::class.java)
    }
}